experimental/skills/DialogToDialog/DialogRootBot/AdapterWithErrorHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.BotBuilderSamples.DialogRootBot.Middleware;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot
+{
+    public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
+    {
+        public AdapterWithErrorHandler(IConfiguration configuration, ICredentialProvider credentialProvider, AuthenticationConfiguration authConfig, ILogger<AdapterWithErrorHandler> logger)
```

Comment:
Should probably align this with the default version in the other samples, such as https://github.com/microsoft/BotBuilder-Samples/blob/4116f005bebe1b5c594ae8623ff816b836432398/samples/csharp_dotnetcore/05.multi-turn-prompt/AdapterWithErrorHandler.cs#L15-L45

Reply:
Not really sure, there is a LOT of code in that sample, I need to think about it.

---
In reply to: [356164475](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356164475) [](ancestors = 356164475)

Reply:
Updated code based on your recommendations (Made some tweaks for the SkillBot so it returns the error and an EndOfConversation to the parent.

---
In reply to: [356416544](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356416544) [](ancestors = 356416544,356164475)